test/dummy/app/controllers/pings_controller.rb

```diff
@@ -7,7 +7,8 @@ def ping
   end
 
   def ping_render
-    response = RPC::DummyAPI::PingResponse.new(double_name: request.name * 2)
+    url = rails_twirp_engine_url
```

Comment:
Where does `rails_twirp_engine_url` come from?

Reply:
It's a regular rails url helper